liferay-echo/liferay-portal
# [TS] LPS-180977 Cannot submit a Form Container form if you would not be able to see any pages when visiting the virtual host directly

Motivation
=======
When submitting a Form via the Form Container Fragment, the user is redirected to the hostname, rather than to the specific page through which they submitted the form. This can cause a problem in the scenario that the user does not actually have permission to see any pages directly from the hostname, which can occur if they are on a page that is hidden from the navigation menu and they do not have permissions to see any other pages on the Site that the hostname represents.

Proposed Solution
========
Add the `p_l_id` parameter to the request for submitting the Form, so that the information about the page is retained during the ThemeDisplay initialization of the received request.

Steps to Verify
========
1. Start up Liferay and log in as the Admin user.
2. Navigate to Control Panel > Objects.
3. Create a new Object Definition named "Test". Add a single field to it, set its Panel Category Key to "Control Panel > Objects", and publish the Object Definition.
4. Navigate to Control Panel > Roles > Guest > Define Permissions.
5. Search for the "Test" Object that you just created, and add the "Add Object Entry" permission to it. Save the changes.
6. Create a new Site named "Test Site".
7. In "Test Site", navigate to Configuration > Site Settings > Site Configuration > Site URL. Fill out the "Virtual Host" field with a valid Virtual Host defined in your hosts file, and click Save.
8. Add a new Widget Page to "Test Site". On the permissions for this Page, remove View permission from the Guest Role.
9. Add a new Content Page to "Test Site". Add a Form Container to the page and select the "Test" Object. Publish the Page.
10. Navigate back to Site Builder and click the three-dot menu for the Content Page you just created > Configure.
11. Enable the "Hidden from Menu Display" slider and click Save.
12. As a Guest user (for example, use an Incognito window, or log out as your admin), navigate to the Content Page that you just created by directly visiting the URL.
13. Try to submit the form.

**Expected Result:** The form would submit successfully.
**Actual Result:** The form fails to be submitted, and a "You do not have permission to view this page." error appears in the UI.
